### ISSUE TITLE
Instructions and config for provisioning platform components

### ISSUE THREAD

**Issue by davgordo** [halkyonio/assemble-platforms#11]
ArgoCD application that includes all platform components so that they can be managed via GitOps.

Acceptance:
- Backstage
- Argo
- Tekton
- Postgres
- Keycloak

**Comment by cmoulliard** [comment]
Can we document to install assemble on k8s and not specifically on ocp ? Does it work on k8s or do we need some ocp components ?

**Comment by davgordo** [comment]
@cmoulliard at the moment, we have a dependency on OLM for operator provisioning. This I think presents a core challenge for interoperability. Otherwise, I think the use of Helm so far is k8s native and neutral. 

We may need to put some thought into ingress in general. I know we're moving in some new directions with OpenShift, so we might take the opportunity to depend less on Routes.

**Comment by cmoulliard** [comment]
> we have a dependency on OLM for operator provisioning

Only ocp uses OLM and operators have not been adopted by the customers except for a few projects. Using helm or carvel packages is a better long term and k8s neutral approach. [Rukpak](https://github.com/operator-framework/rukpak) should be able to help us too

**Comment by cmoulliard** [comment]
> We may need to put some thought into ingress in general. I know we're moving in some new directions with OpenShift, so we might take the opportunity to depend less on Routes.

The platform should take care to install ingress vs OpenShift route or OpenShift DeploymentInfo vs k8s Deployment or  ImageStream vs ...

**Comment by cmoulliard** [comment]
> at the moment, we have a dependency on OLM for operator provisioning

OLM can also be installed on k8s - see [here](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/install/install.md)

**Comment by cmoulliard** [comment]
> we have a dependency on OLM for operator provisioning

Can you describe/detail the depending ?

**Comment by davgordo** [comment]
> The platform should take care to install ingress vs OpenShift route or OpenShift DeploymentInfo vs k8s Deployment or ImageStream vs ...

Good point. At least one of those is actionable already, so I created #23

**Comment by davgordo** [comment]
> OLM can also be installed on k8s - see [here](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/install/install.md)

Yes, I think if we choose to leverage OLM for operator management, and we also support plain k8s Assemble users, it makes sense for us to provide guidance on provisioning OLM. #24

**Comment by davgordo** [comment]
> Can you describe/detail the dependency ?

I only mean to say that so far we have chosen to use OLM for operator lifecycle management, and our manifests for provisioning operators are OLM manifests. That is the extent of the dependency on OLM. I do not mean to suggest there is a unique OLM feature that we depend on.

**Comment by davgordo** [comment]
depends on #21
